rotkehlchen/chain/polygon_pos/node_inquirer.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Literal, cast
+
+from eth_typing import BlockNumber
+from web3 import Web3
+
+from rotkehlchen.chain.constants import DEFAULT_EVM_RPC_TIMEOUT
+from rotkehlchen.chain.evm.contracts import EvmContracts
+from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+from rotkehlchen.chain.evm.types import WeightedNode
+from rotkehlchen.greenlets.manager import GreenletManager
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, SupportedBlockchain, Timestamp
+
+from .constants import POLYGON_POS_ETHERSCAN_NODE, POLYGON_POS_ETHERSCAN_NODE_NAME
+from .etherscan import PolygonPOSEtherscan
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class PolygonPOSInquirer(EvmNodeInquirer):
+
+    def __init__(
+            self,
+            greenlet_manager: GreenletManager,
+            database: 'DBHandler',
+            connect_at_start: Sequence[WeightedNode],
+            rpc_timeout: int = DEFAULT_EVM_RPC_TIMEOUT,
+    ) -> None:
+        etherscan = PolygonPOSEtherscan(
+            database=database,
+            msg_aggregator=database.msg_aggregator,
+        )
+        contracts = EvmContracts[Literal[ChainID.POLYGON_POS]](chain_id=ChainID.POLYGON_POS)
+        super().__init__(
+            greenlet_manager=greenlet_manager,
+            database=database,
+            etherscan=etherscan,
+            blockchain=SupportedBlockchain.POLYGON_POS,
+            etherscan_node=POLYGON_POS_ETHERSCAN_NODE,
+            etherscan_node_name=POLYGON_POS_ETHERSCAN_NODE_NAME,
+            contracts=contracts,
+            connect_at_start=connect_at_start,
+            rpc_timeout=rpc_timeout,
+            contract_multicall=...,  # type: ignore[arg-type]  # TODO: find out and add later
+            contract_scan=...,  # type: ignore[arg-type]  # TODO: find out and add later
+            dsproxy_registry=...,  # type: ignore[arg-type]  # TODO: find out and add later
+        )
+        self.etherscan = cast(PolygonPOSEtherscan, self.etherscan)
+
+    # -- Implementation of EvmNodeInquirer base methods --
+
+    def query_highest_block(self) -> BlockNumber:
+        block_number = self.etherscan.get_latest_block_number()
+        log.debug('Polygon POS highest block result', block=block_number)
+        return BlockNumber(block_number)
+
+    def _is_pruned(self, web3: Web3) -> bool:
+        return False  # TODO: find out a rule for determining pruned nodes in polygon POS
```

Comment:
Both of these _is_pruned and _have_archive should be easy to do. Why not add them? At this point so far it just seems a skeleton PR with lots of TODOs.

The problem is that any future reviews will become **really** hard like this. These 2 TODOs are trivial to implement. Just do them.